packages/react/src/components/hero/Hero.stories.tsx

```diff
@@ -0,0 +1,546 @@
+        },
+      },
+    },
+  };
+};
+
+const DefaultContent = (props: DefaultContentProps) => {
+  const { title, text, buttonTheme } = props;
+  const h1Text = title || 'Welcome to the hero story';
+  const paragraphText = text || defaultText;
+  const blackButtonStyle = { '--background-color': '#000', '--color': '#fff', '--border-color': '#000' };
+  const whiteButtonStyle = { '--background-color': '#fff', '--color': '#000', '--border-color': '#fff' };
+  const buttonStyle = buttonTheme !== 'black' ? whiteButtonStyle : blackButtonStyle;
+  return (
+    <>
+      <Hero.Title>{h1Text}</Hero.Title>
+      <Hero.Text>{paragraphText}</Hero.Text>
+      <Button
+        variant="secondary"
+        role="link"
+        // @ts-ignore
+        style={buttonTheme ? buttonStyle : {}}
+      >
+        Click me
+      </Button>
+    </>
+  );
+};
+
+const NavigationComponent = () => (
+  <Navigation menuToggleAriaLabel="Menu" skipTo="#content" skipToContentLabel="Skip to main content">
+    {/* NAVIGATION ROW */}
+    <Navigation.Row ariaLabel="Main navigation">
+      <Navigation.Item href="#" label="Link" active onClick={(e) => e.preventDefault()} />
+      <Navigation.Item href="#" label="Link" onClick={(e) => e.preventDefault()} />
+      <Navigation.Item href="#" label="Link" onClick={(e) => e.preventDefault()} />
+      <Navigation.Item href="#" label="Link" onClick={(e) => e.preventDefault()} />
+      <Navigation.Dropdown label="Dropdown">
+        <Navigation.Item href="#" label="Link" onClick={(e) => e.preventDefault()} />
+        <Navigation.Item href="#" label="Link" onClick={(e) => e.preventDefault()} />
+        <Navigation.Item href="#" label="Link" onClick={(e) => e.preventDefault()} />
+        <Navigation.Item href="#" label="Link" onClick={(e) => e.preventDefault()} />
+      </Navigation.Dropdown>
+    </Navigation.Row>
+
+    {/* NAVIGATION ACTIONS */}
+    <Navigation.Actions>
+      {/* LANGUAGE SELECTOR */}
+      <Navigation.LanguageSelector label="FI">
+        <Navigation.Item href="#" onClick={(e) => e.preventDefault()} lang="fi" label="Suomeksi" />
+        <Navigation.Item href="#" onClick={(e) => e.preventDefault()} lang="sv" label="På svenska" />
+        <Navigation.Item href="#" onClick={(e) => e.preventDefault()} lang="en" label="In English" />
+        <Navigation.Item href="#" onClick={(e) => e.preventDefault()} lang="fr" label="En français" />
+        <Navigation.Item href="#" onClick={(e) => e.preventDefault()} lang="de" label="Auf deutsch" />
+        <Navigation.Item href="#" onClick={(e) => e.preventDefault()} lang="ru" label="По-русски" />
+      </Navigation.LanguageSelector>
+    </Navigation.Actions>
+  </Navigation>
+);
+
+export const ImageLeft = (args) => {
+  const heroProps: HeroProps = {
+    koros: args.koros,
+    centeredContent: args.centeredContent,
+    theme: { ...imageLeftOrRightTheme, ...args.theme },
+    imageSrc: args.imageSrc || imageFile,
+    variant: 'imageLeft',
+  };
```

Comment:
Stories show each variant, so there should be no need to change the variant, when the story is build just for that variant? `ImageLeft` and `ImageRight` used to be in one story, but separate stories give better user experience, because story controls are not very convenient. And then loki images are created for each variant.

All props, except the variant, can be set in the story controls. The story args are passed to the hero component. The props object is created inside the story, but args are set to it. So user can override them.

```
const heroProps: HeroProps = {
 koros: **args.koros**,
centeredContent: **args.centeredContent,**
 theme: { ...imageLeftOrRightTheme, ...**args.theme** },
 imageSrc: **args.imageSrc** || imageFile,
variant: 'imageLeft',
};
```

[Example](https://city-of-helsinki.github.io/hds-demo/hero/?path=/story/components-hero--image-left&args=centeredContent:true;koros.dense:true;theme.--background-color:!hex(ff9)) For some reason all props are not stored to urls, like imageSrc.

Reply:
Good point, I'll make `variant` editable, so users can see the value immediately. 

The [Playground for theme](https://city-of-helsinki.github.io/hds-demo/hero/?path=/story/components-hero--playground-for-theme) outputs the applied theme under the component so it is easier to copy-paste.

Could that be useful with all props too? A common component that would dynamically show how to use the component with currently applied props?

Reply:
>Could that be useful with all props too? A common component that would dynamically show how to use the component with currently applied props?

The storybook Controls section does that job nicely by default so I don't think it's needed. It's pretty handy to display the theme object though.

Reply:
All properties of all stories are now enabled and there are no props set in the code.